frontend/components/AgentWallet/BalancesAndAssets/BalancesAndAssets.tsx

```diff
@@ -88,11 +89,20 @@ export const BalancesAndAssets = ({
   onLockedFundsWithdrawn,
 }: BalancesAndAssetsProps) => {
   const [isWithdrawModalVisible, setWithdrawModalVisible] = useState(false);
+  const { selectedAgentType } = useServices();
+
+  const handleWithdraw = () => {
+    if (selectedAgentType === AgentMap.PredictTrader) {
```

Comment:
let's better check for `selectedAgentConfig.hasExternalFunds`. if true then `setWithdrawModalVisible`